### PR TITLE
add cleanup function to clean contexts and keys

### PR DESCRIPTION
**Pull request by ArseniyKholod**
This PR adds function `cleanup`, that can be used for runtime cleaning of `CryptoContext`s and generated keys.

It should be merged after https://github.com/hpsc-lab/openfhe-julia/pull/61 and https://github.com/hpsc-lab/OpenFHE.jl/pull/64 and new releases of OpenFHE and openfhe-julia

### CODE REVIEW

File: src/operations.jl
Comment (sloede) [inline src/operations.jl:83]: I'd explicitly return `nothing` instead of whatever the last function called has as its return value.
```suggestion
    OpenFHE.ReleaseAllContexts()
    
    nothing
```

File: src/operations.jl
Comment (sloede) [inline src/operations.jl:67]: I suggest to use a more descriptive name. Maybe something like `clear_contexts` or `release_context_memory`?

**Review by sloede — CHANGES_REQUESTED** [review]
Thanks for figuring this out!

Besides the comments I already left, I have a more general remark:

Wouldn't a better place to hold this `cleanup` function be in OpenFHE.jl? Since it's only relevant for OpenFHE.jl, I feel like it should be located there, as it could also be useful for someone working with OpenFHE.jl without SA.jl.

But this is merely a question at this point, I am not sure myself what the best way to handle this would be.

**Comment by ArseniyKholod** [comment]
> Thanks for figuring this out!
> 
> Besides the comments I already left, I have a more general remark:
> 
> Wouldn't a better place to hold this `cleanup` function be in OpenFHE.jl? Since it's only relevant for OpenFHE.jl, I feel like it should be located there, as it could also be useful for someone working with OpenFHE.jl without SA.jl.
> 
> But this is merely a question at this point, I am not sure myself what the best way to handle this would be.

Thank you for the review!

I agree with you, that this function is both useful for using OpenFHE.jl separately or within SecureArithmetic.jl, I added it here only because it seems, that we try to leave functions in OpenFHE.jl as they are in OpenFHE, which does not provide a single function to clean all the keys, so I thought it fits better in a "user-friendly interface". But it doesn't generally make a big difference for me, where to place it.

File: src/operations.jl
Comment (ArseniyKholod) [reply src/operations.jl:67]: `release_context_memory` sounds well!

**Comment by ArseniyKholod** [comment]
I added docs in README.md to explain how to deal with memory issues.

File: README.md
Comment (sloede) [inline README.md:82]: Please update this section with the updates from https://github.com/hpsc-lab/OpenFHE.jl/pull/64 once we have merged the other PR

File: examples/simple_real_numbers.jl
Comment (sloede) [inline examples/simple_real_numbers.jl:60]:
```suggestion
    release_context_memory()
    GC.gc()
```
Shouldn't we also call GC after each of these calls?

File: src/operations.jl
Comment (sloede) [inline src/operations.jl:75]:
```suggestion
only be freed after restarting the Julia REPL. It is also advisable to call `GC.gc()` after a call to `release_context_memory` to clean up all memory on the Julia side.
```

Comment (sloede) [reply src/operations.jl:75]: don't know if this is correct, please verify

**Review by sloede — CHANGES_REQUESTED** [review]

**Review by sloede — APPROVED** [review]
LGTM!

**Pull request merged**
